### ISSUE TITLE
At top of Panel homepage, the 4 examples return a 503 error. Same in App gallery

### ISSUE THREAD

**Issue by Coderambling** [holoviz/panel#6845]
Observed 503 error at May 17th 2024, 02.19 CET .

Is this a recurring issue?

For example, clicking on Portfolio Anayzer links to
 https://panel-gallery.holoviz.dsp.anaconda.com/portfolio_analyzer

Which returns:

503 Service Temporarily Unavailable

Same for the other three and in App Gallery.

Portfolio Analyzer

￼

Portfolio Optimizer

￼

MRI Slicer

￼

Videostream

￼

Panel

**Comment by philippjfr** [comment]
Will be fixed automatically once we release 1.4.3

**Comment by Coderambling** [comment]
Ah ok. Will that be soon? Having the first 4 links on a homepage but working is not a good look.

**Comment by philippjfr** [comment]
Indeed, we've had a lot of trouble with our deployment infrastructure recently hopefully that'll be a thing of the past. I'll try to release over the weekend.

**Comment by Coderambling** [comment]
AStilla blank for me.

**Comment by Coderambling** [comment]
blank for me.

**Comment by Coderambling** [comment]
And now it's blank again. The other 3 seem to load.

**Comment by Coderambling** [comment]
3 of the 4 examples do not have a link to their source code in the example. Only the one below does. And maybe MRI Slicer but that one is not working...

https://panel-gallery.holoviz-demo.anaconda.com/portfolio_optimizer

**Comment by wybert** [comment]
App in App Gallery can't be reached.
<img width="654" alt="image" src="https://github.com/user-attachments/assets/feac9cc5-51b4-4503-9ec6-df9adcb15fd5">

**Comment by Coderambling** [comment]
1 out of 4 still not working: 

https://panel-gallery.holoviz-demo.anaconda.com/vtk_slicer

@philippjfr , will that be part of the 1 5 milestone?

**Comment by MarcSkovMadsen** [comment]
I've reproduced this today. A user reported on discord that all 4 links gave 404. Only the vtk_slicer give me a 404.

**Comment by Coderambling** [comment]
Same result forme just now.

Update from Discord by @philippjfr :

Really appreciate it, we have to pull those vtk examples for now since our deployment solution doesn't support it.

https://discord.com/channels/1075331058024861767/1270318974650810480/1270443818574942218

**Comment by MarcSkovMadsen** [comment]
Currently I can't find examples in the App Gallery that work. They all give 404. For example

![image](https://github.com/user-attachments/assets/d0089c8f-c675-46fc-ab78-a1ccab642e66)

**Comment by Coderambling** [comment]
Same for me.

**Comment by Coderambling** [comment]
Update: monitoring for this is on the planning to be implemented.

**Comment by Coderambling** [comment]
Update:

All 4 current examples load at the time of this comment. Great!

Additional observations:

There is no light / dark theme toggle in the examples, except for 4.


Example 1.

1.Data for the stocks in 1. ends at jan 2023. Could use a data update, or ideally, an automated periodic update of the stocks data.

2. The 'Reset the Layout' button at the top left does not seem to do anything. After choosing another stock and changing time period, the button does not reset anything.

Or are the layout elements supposed to resizable? I was not able to drag them to change size.

Example 2.

1. Data ends at 2016. Could use a data update, or ideally, an automated periodic update of the stocks data.
2. Clicking on a  stock in the top right-hand stock table results in the busy indicator running briefly, but it does not result in any change in the UI. Is that supposed to be the case?
Multi-select using Ctrl-mouse click result in multiple stocks being highlighted, but again, nothing happens. Is multi-select supposed to be possible, and supposed to result in a change?

I.e. if you select a subset of the 4 stocks, are the portfolio information graphs supposed to be updated using only that subset? That would make sense / be useful I guess, but is that the intended functionality?

Or should the stocks not be selectable?

Example 3.

Would a progress indicator of the model that is loading be viable as an enhancement of the example? Would be nice, and make sense, especially for loading of the larger models.

Also, a ready (succesfully finished loading) indicator would be nice, maybe together with a message in the chat window that the model has loaded?

After loading 1 model, it seems to not be possible to load another model. Another model can be chosen from the dropdown, but the Load model button stops working after the first loading of a model.

A loaded model does not give responses, other the stating the model name, i.e. 'TinyLlama-1.1B-Chat-v1.0-q4f16_1-MLC-1k' .

Example 4.

Is the videostream supposed to activate the camera of the local device and use that camera as the videostream source?







1. Portfolio Analyzer
Portfolio Analyzer App
https://panel-gallery.holoviz-demo.anaconda.com/portfolio_analyzer

2. Portfolio Optimizer
Portfolio Optimizer App
https://panel-gallery.holoviz-demo.anaconda.com/portfolio_optimizer

3. WebLLM Chat App
WebLLM Application
https://panel-gallery.holoviz-demo.anaconda.com/webllm

4. Videostream
Videostream App
https://panel-gallery.holoviz-demo.anaconda.com/streaming_videostream

**Comment by rekognizecode** [comment]
WebLLM now shows a progress bar including what is currently happening, eg loading the model, copying parameters, ...  
When the model was loaded correctly, it shows _Loaded TinyLlama-1.1B-Chat-v1.0-q4f16_1-MLC-1k, start chatting!_, or similar.  
What is not nice, when the model isn't loaded correctly, no error message gets shown to the user, even though the developer console shows errors, which could be used to show some errors to the user.  
As you mentioned, the model responds with it's own name, but in my experience only when the model isn't loaded properly.

**Comment by Coderambling** [comment]
Thanks for the update @rekognizecode ! If possible, could you describe / paste / do you have a screenshot other developer console errors? @ahuang11 , do you know if there is a current failure mode / error messsage in the component when loading of models fails / there is a time out?

**Comment by rekognizecode** [comment]
Very welcome!


![Image](https://github.com/user-attachments/assets/bf2c6f2d-ab16-47bd-98e6-310086954855)
<i>Developer console showing `ModelNotFoundError` when trying to load SmolLM</i>  
This error shows in every browser I tested, seems to be a configuration error, or the model is simply not available anymore.
I [opened an issue](https://github.com/holoviz/panel/issues/7957) about this   

![Image](https://github.com/user-attachments/assets/810d9170-6fe6-4209-abeb-3c0b6e81558d)
<i>Developer console showing `WebGPUNotAvailableError` in Firefox</i>
  This error currently shows on all of the models in Firefox (except SmolLM where the beforementioned error shows) because WebGPU is not supported yet, in Chrome these models work without problems.
A colleague got a similar error on his Apple Silicon MacBook, either that Apple M-Chips are not supported or that Safari doesn't support WebGPU yet, not sure.

**Comment by ahuang11** [comment]
That webllm code is outdated I think. Maybe try:

https://panel-extensions.github.io/panel-web-llm/

**Comment by rekognizecode** [comment]
@ahuang11 Then this version should be linked on Panel's landing page instead :D

But to be honest, the site you linked seems quite unfinished, when I try to load even a small model, it only shows a loading indicator on the download button and nothing happens.
JS console is full of stuff, partly Python code being logged, but no helpful error messages sadly.

EDIT: @ahuang11 Did you manage to get the version at https://panel-extensions.github.io/panel-web-llm/ to work?

**Comment by philippjfr** [comment]
Going to close. I've fixed the WebLLM issues. Apart from that:

>There is no light / dark theme toggle in the examples, except for 4.

Not all examples have to have a theme toggle.

>The 'Reset the Layout' button at the top left does not seem to do anything. After choosing another stock and changing time period, the button does not reset anything.

It resets the layout, should you re-arrange the plots, it does not reset the widgets.

>Data ends at 2016. Could use a data update, or ideally, an automated periodic update of the stocks data.

I see no reason to do this really.

>Clicking on a stock in the top right-hand stock table results in the busy indicator running briefly, but it does not result in any change in the UI. Is that supposed to be the case?

It's a form, you have to confirm with the button.

>Would a progress indicator of the model that is loading be viable as an enhancement of the example? Would be nice, and make sense, especially for loading of the larger models.

There is a loading indicator.

>Also, a ready (succesfully finished loading) indicator would be nice, maybe together with a message in the chat window that the model has loaded?

Same.

>Is the videostream supposed to activate the camera of the local device and use that camera as the videostream source?

Yes.

**Comment by philippjfr** [comment]
>Clicking on a stock in the top right-hand stock table results in the busy indicator running briefly, but it does not result in any change in the UI. Is that supposed to be the case?

> It's a form, you have to confirm with the button.

Correcting myself, we removed the button. The calculation just takes a second.

**Comment by Coderambling** [comment]
VTK Slicer works now (it is not one of the 4 apps at the top anymore).

Regarding your point below. Using an older dataset will simple make the app feel increasingly outdated when a new user comes across it. So no technical reason, but related to the general experience.

Data ends at 2016. Could use a data update, or ideally, an automated periodic update of the stocks data.

> I see no reason to do this really.
